### PR TITLE
Update README.rst and add tuttest pipelines to validate commands in the README.rst file

### PR DESCRIPTION
**Pull request by joannabrozek**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -19,9 +20,11 @@ Build required binaries
 You can build all required binaries using provided ``build_binaries.sh`` script. This script will build Surelog, Yosys and the systemverilog-plugin and place them into ``image`` folder. You need to add this folder into your ``PATH`` variable to make sure you are using correct versions of the binaries.
 
 .. code-block:: bash
+   :name: build-binaries
 
    #Make sure submodules are inited and updated to the latest version
    git submodule update --init --recursive
+   pip3 install orderedmultidict
```
Comment (tgorochowik) [inline README.rst:27]: This should probably be moved above, to the dependencies block

File: README.rst [hunk 2]
```diff
@@ -32,17 +35,23 @@ Usage
...
 
 .. code-block:: bash
+   :name: show-help
 
-   yosys
-   plugin -i systemverilog
-   help read_systemverilog
-   help read_uhdm
+   yosys -p "plugin -i systemverilog" -p "help read_systemverilog"
+   yosys -p "plugin -i systemverilog" -p "help read_uhdm"
```
Comment (tgorochowik) [inline README.rst:54]: Can we keep this help with the regular interactive yosys flow?

Then we should be able to use tuttest like this:
```
yosys -s <(tuttest README.rst show-help)
```

File: README.rst [hunk 3]
```diff
@@ -2,15 +2,57 @@ yosys-uhdm-plugin-integration
...
+
+Before installing the plugin, check that yosys is installed and correctly configured:
+
+.. code-block:: bash
+   :name: check-yosys
+
+   yosys -version
+   yosys-config --help
+
+If you don't have yosys, for Debian-based Linux distributions you can install it (in case of Ubuntu, you need Ubuntu 22.04 LTS or higher):
```
Comment (tgorochowik) [inline README.rst:16]: Can we have two separate sections, one for debian:sid (where it works from the built-in repos) and one from Ubuntu where we download this package with wget?

We should also have a quick introduction in which we mention the minimal required yosys version for the plugin to work

File: README.rst [hunk 4]
```diff
@@ -2,15 +2,57 @@ yosys-uhdm-plugin-integration
...
+
+   apt install -y curl jq tar
+   curl https://api.github.com/repos/antmicro/yosys-uhdm-plugin-integration/releases/latest | jq .assets[1] | grep "browser_download_url" | grep -Eo 'https://[^\"]*' | xargs wget -O - | tar -xz
+
+After downloading the plugin, the next step is plugin installation:
+
+.. code-block:: bash
+   :name: install-plugin
+
+   ./install_plugin.sh
```
Comment (tgorochowik) [inline README.rst:41]: will it work without sudo for regular users?

**Review by tgorochowik — COMMENTED** [review]
Some minor comments.

Also - can we have a single tuttest.yml with two jobs in it? They are pretty similar so it makes sense to group them in a single file

File: README.rst [hunk 5]
```diff
@@ -3,15 +3,65 @@ SystemVerilog support for Yosys
...
+
+Before installing the plugin, check that yosys is installed and correctly configured:
+
+.. code-block:: bash
+   :name: check-yosys
+
+   yosys -version
+   yosys-config --help
+
+The required yosys version is 0.18 or later.
```
Comment (tgorochowik) [inline README.rst:17]: Isn't it `0.10` or later? @kamilrakoczy

Comment (kamilrakoczy) [reply README.rst:17]: Yes, missing ``Pass::on_register`` should be available from ``0.10``: https://github.com/YosysHQ/yosys/commit/cd92a974f4cf8d4db74d504c38e51ce043e02403

File: .github/workflows/tuttest-developer.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  pull_request:
+
+jobs:
+
+  test-plugin-from-sources:
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:jammy
+    env:
+      CC: gcc-9
+      CXX: g++-9
```
Comment (kamilrakoczy) [inline .github/workflows/tuttest-developer.yml:16]: I think we should remove this variables, or move them to the README.

**Review by tgorochowik — APPROVED** [review]

**Pull request merged**